### PR TITLE
Revisit Docs 2

### PR DESCRIPTION
**Pull request by welldan97**
Update documentation for 4, 6, 7, 8, and improve overall structure and formatting

### CODE REVIEW

File: book/guide/0-prerequisites.md [hunk 1]
```diff
@@ -9,68 +10,71 @@
...
 
 ## Prerequisites for the OPUS Pool SDK Example
 
-Before we dive into the core of our tutorial, it's essential to understand the prerequisites and the foundation on which we're building. This guide is centered around the OPUS Pool SDK Example. 
+Before we dive into the core of our tutorial, it’s essential to understand the prerequisites and the foundation on which we’re building. This guide is centered around the OPUS Pool SDK Example.
+
+The repository for this example can be found [here][opus-pool-example].
 
-The repository for this example can be found [here][opus-pool-example]. 
+Please review the README in the repository for instructions on installation and how to run the project.
```
Comment (mksh) [inline book/guide/0-prerequisites.md:34]: Would be better read as "Please review the README file"

File: book/guide/1-installation-and-setup.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+-   [Using Vaults with OPUS Pool SDK](#using-vaults-with-opus-pool-sdk)
+    -   [Option 1: Accessing Default Vault Address](#option-1-accessing-default-vault-address)
+    -   [Option 2: Using Custom Vault Addresses](#option-2-using-custom-vault-addresses)
+-   [Configuring wagmi in Your Application](#configuring-wagmi-in-your-application)
+-   [Utilizing wagmi Hooks for Wallet Interactions](#utilizing-wagmi-hooks-for-wallet-interactions)
+-   [Next Steps](#next-steps)
+
+## Installation
+
+To begin integrating the OPUS SDK into your project, the first step is to add the library to the application. You can use your favorite package manager, including npm, yarn or pnpm. For simplicity, we will use npm.
```
Comment (mksh) [inline book/guide/1-installation-and-setup.md:13]: Probably worth mentioning `bun`, to make more appeal to its fans :)

File: book/guide/1-installation-and-setup.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+This function will return a list of default vault addresses, allowing you to fully leverage the capabilities of the OPUS PoolSDK in your project.
+
+### Option 2: Using Custom Vault Addresses
+
+The OPUS Pool SDK supports this flexibility if you prefer to use custom vaults and have specific vault addresses for your project. You can configure these addresses as part of your setup, enabling you to work with vaults of your choice.
+
+# Integrating the wagmi Library
+
+We will use the wagmi and web3modal libraries to connect with Ethereum wallets, allowing users to interact with blockchain applications. This is an essential step, particularly because the OPUS Pool SDK does not provide wallet connection functionality.
```
Comment (mksh) [inline book/guide/1-installation-and-setup.md:96]: > the wagmi and web3modal

Adjective likely not necessary here

File: book/guide/2-fetching-vault-details.md [hunk 4]
```diff
@@ -90,16 +90,14 @@ const Vault = ({ vaultDetails }: { vaultDetails: VaultDetails[] }) => {
 };
 ```
 
-Once rendered, the vault details will be displayed in an informative table on the screen, showcasing key information such as vault name, APY, and balances in a user-friendly format:
+Once rendered, the vault details will be displayed in an informative table on the screen, showcasing key information such as vault name, APY, and balances in a user-friendly format:
```
Comment (mksh) [inline book/guide/2-fetching-vault-details.md:93]: Probably don't need article here 

> in a user friendly format
> in user-friendly format

File: book/guide/3-staking.md [hunk 5]
```diff
@@ -50,11 +50,11 @@ export const FormComponent = ({ onSubmit }) => {
 };
 ```
 
-This form component uses the `useAccount` and `useBalance` hooks from `wagmi` to manage user account details and balance information. The `onSubmit` prop is a function that handles the actual staking process, which you can define in your application logic.
+This form component uses the `useAccount` and `useBalance` hooks from `wagmi` to manage user account details and balance information. The `onSubmit` prop is a function that handles the actual staking process, which you can define in your application logic.
 
 {% hint style="info" %}
 
-For a more comprehensive implementation, which includes error handling, loading states, and styling, refer to our [extended version][stake-ui], which is available in the codebase.
+For a more comprehensive implementation, which includes error handling, loading states, and styling, refer to our [extended version][stake-ui], which is available in the codebase.
```
[the remark this answers is not in the document]
Comment (mksh) [reply book/guide/3-staking.md:57]: for the more comprehensive implementation

File: book/guide/8-rewards-history.md [hunk 6]
```diff
@@ -1,24 +1,28 @@
...
 
-In this chapter, we'll focus on fetching and visualizing the rewards history using the Opus SDK. The SDK offers a `getRewardsHistory` method, which retrieves the history of rewards earned from specific vaults.
+-   [Overview](#overview)
+-   [Initializing the Pool and Fetching Rewards History](#initializing-the-pool-and-fetching-rewards-history)
+-   [Visualizing Rewards Data with a Chart](#visualizing-rewards-data-with-a-chart)
+-   [Chart Representation on the Screen](#chart-representation-on-the-screen)
+
+## Overview
+
+In this chapter, we’ll explore the process of fetching and visualizing rewards history using the Opus SDK. Visualizing reward data helps users gain insights into their earnings over time and make informed decisions about their staking activities.
```
Comment (mksh) [inline book/guide/8-rewards-history.md:10]: Replace

> in this chapter

with

> in this section

File: book/guide/0-prerequisites.md (hunk 1 above)
Comment (welldan97) [reply book/guide/0-prerequisites.md]: fixed

File: book/guide/1-installation-and-setup.md (hunk 3 above)
Comment (welldan97) [reply book/guide/1-installation-and-setup.md]: fixed

File: book/guide/2-fetching-vault-details.md (hunk 4 above)
Comment (welldan97) [reply book/guide/2-fetching-vault-details.md]: fixed

File: book/guide/3-staking.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (welldan97) [reply book/guide/3-staking.md]: fixed

File: book/guide/8-rewards-history.md (hunk 6 above)
Comment (welldan97) [reply book/guide/8-rewards-history.md]: fixed, thank you!

File: book/guide/1-installation-and-setup.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+const { address } = useAccount();
+
+// Obtain a client to query public blockchain data
+const publicClient = usePublicClient();
+
+// Use the walletClient hook to execute transactions
+const { data: walletClient } = useWalletClient();
+```
+
+In this example, `useAccount` is used to obtain the user’s wallet address; the `usePublicClient` hook provides a client for querying public blockchain data. Finally, the `useWalletClient` hook provides a client for executing transactions, enabling interactions with the blockchain.
```
[the remark this answers is not in the document]
Comment (mksh) [reply book/guide/1-installation-and-setup.md:152]: Ok, makes sense

**Review by mksh — APPROVED** [review]
LGTM

**Pull request merged**